_posts/2022-04-04-valheim-server.md

```diff
@@ -0,0 +1,130 @@
+author: Joseph Miocevich
+image: img/valheim-logo.webp
+description:
+  How to set up a dedicated server for Valheim on AWS, with some discord notifications
+---
+
+![Header](/img/valheim-logo.webp)
+
+Do you want to set up your own Valheim Server? Are you sick of waiting for your friend to figure out how to port forward their local server? Has your wallet been raided by an 'easy' online game hosting solution? Then this is the blog post for you! Read on if you want to become a certified Viking nerd.
+
+Now, if you are like me, where I decide 9 pm is when I should do that documentation I've been procrastinating on, the hours that my friends play and I play rarely aligns, hence the need for a dedicated server. I’ve seen a few online tutorials on how to set up your own Valheim Server, but I took it as an opportunity to do what any semi self-loathing programmer does.... ignore it all and decide I can do better...I mean attempt to do it myself as a learning opportunity.
+
+If you just want the code, here you go: [ Valheim-Server-on-AWS-Part-1 ](https://github.com/JMiocevich/Valheim-Server-on-AWS-Part-1)
+
+We are gonna separate this post into two parts. Part 1 will be the bare minimum to get a server up and running, whilst part 2 will include some fancier bells and whistles.
+
+So let's cover some basic criteria on what creating your own virtual Vikings adventure land will entail:
+
+- Infrastructure as code using AWS SAM (minimal clicksops)
+- Easy deployment for the technically challenged
+- Secure and resilient architecture
+- Cost competitive against out of the box services
+- Cheap
+- Ability to Auto stop server when game world is not in use
+- Get Notifications on the game state in discord (server start/stop/when a player joins)
+- Automatic Backup
+- Did I mention cheap?
+
+Part 2:
+- Ability to send server commands through discord (start/stop/ban players)
+- Edit server settings through discord
+- Automatic IP address retrieval through discord
+
+***
+<br/>
+## Instructions
+
+The following/requirements are needed.
+- Basic understanding of nodeJS
+- Access to AWS account with access credentials avaliable for use,  [AWS Docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+- [Visual Studio Code](https://code.visualstudio.com) installed
+- [Node.js](https://nodejs.org/en/download/) installed
+- [AWS Command Line Interface](https://aws.amazon.com/cli/) installed
+- [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) installed
+- Following github repo downloaded [Valheim Server](https://github.com/JMiocevich/valheim_server_aws_ts)
+- [Discord WebHook](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks)
+
+See [ setup instructions ](https://github.com/JMiocevich/Valheim-Server-on-AWS-Part-1/blob/main/README.md) for step by step instructions on setup.
+
+## Discussion and Reasoning
+
```

Comment:
These points underline why an architecture diagram is important.

You haven't outlined anywhere else in the post what your solution is - here you are jumping into your design decisions without introducing what they are (don't make me go off and read the code).